File: carbon/src/main/java/carbon/component/BottomSheetRow.java
package carbon.component;

import android.view.ViewGroup;

import carbon.R;
import carbon.beta.BottomSheetLayout;
import carbon.databinding.CarbonBottomsheetRowBinding;

public class BottomSheetRow extends LayoutComponent<BottomSheetLayout.Item> {

    private final CarbonBottomsheetRowBinding binding;

    public BottomSheetRow(ViewGroup parent) {
        super(parent, R.layout.carbon_bottomsheet_row);
        binding = CarbonBottomsheetRowBinding.bind(getView());
    }

    @Override
    public void bind(BottomSheetLayout.Item data) {
        super.bind(data);
        binding.carbonItemIcon.setImageDrawable(data.getIcon());
        if (data.getIconTintList() != null)
            binding.carbonItemIcon.setTintList(data.getIconTintList());
        binding.carbonItemText.setText(data.getTitle());
    }

}

File: carbon/src/main/java/carbon/component/BottomSheetRow.kt
package carbon.component

import android.view.ViewGroup
import carbon.R
import carbon.beta.BottomSheetLayout
import carbon.databinding.CarbonBottomsheetRowBinding

class BottomSheetRow(parent: ViewGroup) : LayoutComponent<BottomSheetLayout.Item>(parent, R.layout.carbon_bottomsheet_row) {
    private val binding = CarbonBottomsheetRowBinding.bind(view)

    override fun bind(data: BottomSheetLayout.Item) {
        binding.carbonItemIcon.setImageDrawable(data.icon)
        if (data.iconTintList != null)
            binding.carbonItemIcon.setTintList(data.iconTintList)
        binding.carbonItemText.text = data.title
    }
}